clusterloader2/docs/network-perf-testcase-enhancement/README.md

```diff
@@ -0,0 +1,286 @@
+nitty-gritty.
+-->
+
+### Test Scenarios
+
+1. Baseline Testcases
+
+	Following testcases will be able to gather the baseline values for network performance metrics. 
+
+	1.1 Point-To-Point (P2P)
+
+	  ![P2P](p2p.PNG)
+
+	  1.1.1 Client-Server POD on different worker node (1:1)
+
+	  Testcase-Id | Testcase Description | Tool | Metrics Measured
+	  ----------- | ---------------------|------|-----------------
+	  TC1.1.1.1   | Send TCP traffic from client pod IP to server pod IP | iperf3 | Throughput in Gbits/s
+	  TC1.1.1.2   | Send UDP traffic from client pod IP to server pod IP | iperf2 | Packets per second (pps), Loss Rate, Latency, jitter
+	  TC1.1.1.3   | Send HTTP traffic from client pod IP to server pod IP | siege | Latency , ResponseTime
+
+
+	1.2 ClusterIP / NodePort Services
+
+	  ![svc](svc.PNG)
+
+	  1.2.1 Client-Server POD on different worker node (1:1)
+
+	  TestcaseId | Testcase Description | Tool | Metrics Measured
+	  ----------- | ---------------------|------|-----------------
+	  TC1.2.1.1   | Send TCP traffic from client pod IP to ClusterIP. 1 server pod is located behind ClusterIP service. | iperf3 | Throughput in Gbits/s
+	  TC1.2.1.2   | Send UDP traffic from client pod IP to ClusterIP. 1 server pod is located behind ClusterIP service. | iperf2 | Packets per second (pps), Loss Rate, Latency, jitter
+	  TC1.2.1.3   | Send HTTP traffic from client pod IP to ClusterIP. 1 server pod is located behind ClusterIP service. | siege | Latency , ResponseTime
+	  TC1.2.1.4   | Send TCP traffic from client pod IP to NodePort svc IP. 1 server pod is located behind NodePort service. | iperf3 | Throughput in Gbits/s
+	  TC1.2.1.5   | Send UDP traffic from client pod IP to NodePort svc IP. 1 server pod is located behind NodePort service. | iperf2 | Packets per second (pps), Loss Rate, Latency, jitter
+	  TC1.2.1.6   | Send HTTP traffic from client pod IP to NodePort svcIP. 1 server pod is located behind NodePort service. | siege | Latency , ResponseTime
+
+
+2. Scale Testcases
+
+    2.1 Point-To-Point (P2P) 
+
+    2.1.1 Client-Server POD on different worker node (N:1)
+
+	  TestcaseId  | Testcase Description | Tool | Metrics Measured
+	  ----------- | ---------------------|------|-----------------
+	  TC2.1.1.1   | Send TCP traffic from N=50 client pods to 1 server pod IP. | iperf3 | Throughput in Gbits/s
+	  TC2.1.1.2   | Send TCP traffic from N=100 client pods to 1 server pod IP. | iperf3 | Throughput in Gbits/s
+	  TC2.1.1.3   | Send UDP traffic from N=50 client pods to 1 server pod IP. | iperf2 | Packets per second (pps), Loss Rate, Latency, jitter
+	  TC2.1.1.4   | Send UDP traffic from N=50 client pods to 1 server pod IP. | iperf2 | Packets per second (pps), Loss Rate, Latency, jitter
+	  TC2.1.1.5   | Send HTTP traffic from N=50 client pods to 1 server pod IP. | siege | Latency , ResponseTime
+	  TC2.1.1.6   | Send HTTP traffic from N=100 client pods to 1 server pod IP. | siege | Latency , ResponseTime
```

Comment:
N=50 ?

[as everywhere else]

Reply:
For TC2.1.1.4 , N=50 is typo error. We want to run both N=50 and N=100 clients for all 3 protocols TCP/UDP/HTTP. Right now these testcases (section 2.1.1 ) are not yet implemented. So only when we run we may know the difference. After that we can take decision whether to take both or only one of them.